bot/bot.go

```diff
@@ -123,12 +123,21 @@ func (b *Bot) loadChannels() (map[string]string, error) {
 // load the public channels and list of all users from current space
 func (b *Bot) loadSlackData() error {
 	// whitelist users by group
-	for _, groupName := range b.config.Slack.AllowedGroups {
-		group, err := b.slackClient.GetUserGroupMembers(groupName)
-		if err != nil {
-			return errors.Wrap(err, "error fetching user of group. You need a user token with 'usergroups:read' scope permission")
+	allGroups, err := b.slackClient.GetUserGroups()
```

Comment:
please call this `b.config.Slack.AllowedGroups` only, if some `AllowedGroups` are defined.
Reason: now it would also call the endpoint if you just want a user-based whitelist. And this `GetUserGroupMembers` will fail, if the scope `usergroups:read` is missing (which requires a paid plan) -> bot would not start in this cases.

Reply:
Thanks for giving me a feedback. I fixed it to avoid making the bot being crashed by adding users in groups but printing warning message.